src/components/HdResponsive.vue

```diff
@@ -18,42 +18,46 @@ export default {
     return {
       matches: {},
       indeterminate: !matchMediaAvailable,
+      listeners: {},
     };
   },
   watch: {
     breakpoints() {
       this.setResponsiveFlags();
     },
   },
-  created() {
+  beforeMount() {
     this.setResponsiveFlags();
   },
-  mounted() {
-    this.addResizeEvents();
-  },
-  beforeDestroy() {
-    this.removeResizeEvents();
+  created() {
+    this.setResponsiveFlags();
   },
   methods: {
+    updateMatch(key, value) {
+      this.matches[key] = value;
+      this.$forceUpdate();
+    },
     setResponsiveFlags() {
+      this.listeners = {};
+
       const breakpoints = {
         ...getBreakpoints(),
         ...this.breakpoints,
       };
 
-      this.matches = Object.keys(breakpoints).reduce(
-        (matches, breakpointKey) => ({
-          ...matches,
-          [breakpointKey]: mediaMatches(breakpoints[breakpointKey]),
-        }),
-        {},
-      );
-    },
-    addResizeEvents() {
-      onResize.onThrottled(this.setResponsiveFlags);
-    },
-    removeResizeEvents() {
-      onResize.offThrottled(this.setResponsiveFlags);
+      for (const breakpointKey in breakpoints) {
+        this.listeners[breakpointKey] = window.matchMedia(breakpoints[breakpointKey]);
+        this.matches[breakpointKey] = this.listeners[breakpointKey].matches;
+        try {
+          this.listeners[breakpointKey].addEventListener('change', (e) => {
+            this.updateMatch(breakpointKey, e.matches);
+          });
+        } catch (error) {
+          this.listeners[breakpointKey].addListener((e) => {
+            this.updateMatch(breakpointKey, e.matches);
+          });
+        }
+      }
```

Comment:
To avoid this `for...in` I would go for something like `Object.entries(breakpoints).forEach(([key, value]) => {...})` so you can remove the ESLint ignore rules as well.

Doing `breakpoints[breakpointKey]` in the middle of the code is quite hard to read. So I would consider extracting it to variable or the solution I mentioned also touches it.

Reply:
👍  Als using that notation would remove the need of having `breakpoints[breakpointKey]` because we now have `value`